addons/website_slides_survey/static/src/js/slides_upload.js

```diff
@@ -76,17 +84,46 @@ SlidesUpload.SlideUploadDialog.include({
      * @override
      * @private
      */
-    _getSelect2DropdownValues: function (){
+    _getSelect2DropdownValues: function () {
         var result = this._super.apply(this, arguments);
 
         var certificateValue = this.$('#certification_id').select2('data');
+        var survey = {};
         if (certificateValue) {
-            result['survey_id'] =  certificateValue.id;
+            if (certificateValue.create) {
+                survey.id = false;
+                survey.title = certificateValue.text;
+            } else {
+                survey.id = certificateValue.id;
+            }
         }
+        result['survey'] = survey;
         return result;
-    }
-});
+    },
+
+    _onFormSubmitDone: function (data) {
+        if (!data.error && data.is_new_survey) {
+            sessionStorage.setItem("certification_toast", data.redirect_url);
+            window.location.reload();
+        }
+        else {
+            this._super.apply(this, arguments);
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
 
+   /**
+    * Will automatically set the title of the slide to the title of the chosen certification
+    */
+    _onChangeCertification: function (ev) {
+        if (ev.added) {
+            this.$("input#name").val(ev.added.text);
```

Comment:
Could get the value through `this.$('input#certification_id')` instead of that strange ev.edded thing?

Reply:
Actually it is due to select2 I think. Easier to do like that.